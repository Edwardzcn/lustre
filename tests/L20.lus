
node imported sqrt (x : real) returns (y : real);
/*@contract
    assume x >= 0.0;
    guarantee y >= 0.0 and y * y = x;
*/

node test (i : real) returns (j : real)
let
  j = sqrt(i);
tel



