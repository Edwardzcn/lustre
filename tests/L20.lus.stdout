PhaseToCore
===========

node (i : real)
  returns (j$u5)
  assumes (Implies)
  shows (Geq)
let
  x : real
    = i
   
  Implies : bool
    = Implies(Geq(x, 0.0), And(Geq(j$u5, 0.0), Eq(Mul(j$u5, j$u5), x)))
   
  Geq : bool
    = Geq(x, 0.0)
tel

--- Step 1 ---
i : real  = 1.0
j = nil
--- Step 2 ---
i : real  = 2.0
j = nil
--- Step 3 ---
i : real  = (EOF)
